src/btree/iter.rs

```diff
@@ -155,17 +251,42 @@ impl<'a> BTreeIterator<'a> {
 		record_id: u64,
 		col: &BTreeTable,
 		log: &impl LogQuery,
-	) -> Result<Option<(Vec<u8>, Vec<u8>)>> {
+	) -> Result<Option<(Vec<u8>, Value)>> {
 		let BtreeIterBackend(tree, iter) = &mut self.iter;
-		if record_id != tree.record_id {
+		if record_id != tree.record_id || self.direction != IterDirection::Forward {
```

Comment:
Isn't change of direction handle at backend iterator level already (BTreeIterState)? (commit overlay should not need it)

Reply:
Decided to update its state here and remove `BTreeIterState`'s direction update.

Otherwise, we would need to also pass `from_seek` to `BTreeIterState::next`, which would make more cubersome.